src/main/scala/org/deepdive/ddlog/DeepDiveLogPrettyPrinter.scala

```diff
@@ -93,36 +95,52 @@ object DeepDiveLogPrettyPrinter extends DeepDiveLogHandler {
       case NegationCond(c) => s"[!${printCond(c)}]"
       case CompoundCond(lhs, op, rhs) => {
         op match {
-          case LogicOperator.AND => s"[${printCond(lhs)}, ${printCond(rhs)}]" 
+          case LogicOperator.AND => s"[${printCond(lhs)}, ${printCond(rhs)}]"
           case LogicOperator.OR  => s"[${printCond(lhs)}; ${printCond(rhs)}]"
         }
       }
     }
   }
 
-  def print(cq: ConjunctiveQuery): String = {
-    val printAtom = {a:Atom =>
-      val vars = a.terms map printExpr
-      s"${a.name}(${vars.mkString(", ")})"
+  def printAtom(a: Atom) = {
+    val vars = a.terms map printExpr
+    s"${a.name}(${vars.mkString(", ")})"
+  }
+
+  def printQuantifiedBody(a: QuantifiedBody) : String = {
+    val modifier = a.modifier match {
+      case ExistModifier(negated) => if(negated) "NOT " else "" + "EXISTS"
+      case OuterModifier() => "OPTIONAL"
+      case AllModifier() => "ALL"
+    }
+    val bodyStr = (a.bodies map printBody).mkString(", ")
+    s"${modifier}[${bodyStr}]"
+  }
+
+  def printBody(b: Body) = {
+    b match {
+      case b: Atom => printAtom(b)
+      case b: Cond => printCond(b)
+      case b: QuantifiedBody => printQuantifiedBody(b)
```

Comment:
It may be okay to name them all `print` since the parameter type is clear enough.